pkg/ipmasq/cell/cell_test.go

```diff
@@ -0,0 +1,116 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package cell
+
+import (
+	"context"
+	"io"
+	"log/slog"
+	"testing"
+
+	"github.com/cilium/cilium/pkg/ipmasq"
+
+	upstreamHive "github.com/cilium/hive"
+	"github.com/cilium/hive/cell"
+	"github.com/cilium/hive/hivetest"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/cilium/cilium/pkg/hive"
+	ipmasqmaps "github.com/cilium/cilium/pkg/maps/ipmasq"
+	"github.com/cilium/cilium/pkg/metrics"
+	"github.com/cilium/cilium/pkg/metrics/metric"
+	"github.com/cilium/cilium/pkg/option"
+)
+
+func TestIPMasqAgentCell(t *testing.T) {
+	option.Config.EnableIPMasqAgent = true
```

Comment:
If you move `option.EnableIPMasqAgent` to the Cell, you can also remove this line and can set the flag in `hive.AddConfigOverride` instead.